crates/bevy_winit/src/lib.rs

```diff
@@ -41,195 +47,49 @@ pub struct WinitPlugin;
-
-                        window.set_outer_position(PhysicalPosition {
-                            x: screen_size.width.saturating_sub(window_size.width) as f64 / 2.
-                                + monitor.position().x as f64,
-                            y: screen_size.height.saturating_sub(window_size.height) as f64 / 2.
-                                + monitor.position().y as f64,
-                        });
-                    } else {
-                        warn!("Couldn't get monitor selected with: {monitor_selection:?}");
-                    }
-                }
-                bevy_window::WindowCommand::SetResizeConstraints { resize_constraints } => {
-                    let window = winit_windows.get_window(id).unwrap();
-                    let constraints = resize_constraints.check_constraints();
-                    let min_inner_size = LogicalSize {
-                        width: constraints.min_width,
-                        height: constraints.min_height,
-                    };
-                    let max_inner_size = LogicalSize {
-                        width: constraints.max_width,
-                        height: constraints.max_height,
-                    };
+        {
+            let (commands, event_loop, new_windows, winit_windows) =
+                system_state.get_mut(&mut app.world);
 
-                    window.set_min_inner_size(Some(min_inner_size));
-                    if constraints.max_width.is_finite() && constraints.max_height.is_finite() {
-                        window.set_max_inner_size(Some(max_inner_size));
-                    }
-                }
-                bevy_window::WindowCommand::Close => {
-                    // Since we have borrowed `windows` to iterate through them, we can't remove the window from it.
-                    // Add the removal requests to a queue to solve this
-                    removed_windows.push(id);
-                    // No need to run any further commands - this drops the rest of the commands, although the `bevy_window::Window` will be dropped later anyway
-                    break;
-                }
-            }
-        }
-    }
-    if !removed_windows.is_empty() {
-        for id in removed_windows {
-            // Close the OS window. (The `Drop` impl actually closes the window)
-            let _ = winit_windows.remove_window(id);
-            // Clean up our own data structures
-            windows.remove(id);
-            window_close_events.send(WindowClosed { id });
+            // Here we need to create a winit-window and give it a WindowHandle which the renderer can use.
+            // It needs to be spawned before the start of the startup-stage, so we cannot use a regular system.
+            // Instead we need to create the window and spawn it using direct world access
+            create_window_system(commands, &**event_loop, new_windows, winit_windows);
         }
+
+        system_state.apply(&mut app.world);
```

Comment:
```suggestion
       // Write changes to world
        system_state.apply(&mut app.world);
```

Not sure this is useful or helpful